src/main/java/appeng/tile/crafting/TileCraftingStorageTile.java

```diff
@@ -36,24 +38,29 @@ protected ItemStack getItemFromTile( final Object obj )
 		final IBlocks blocks = AEApi.instance().definitions().blocks();
 		final int storage = ( (TileCraftingTile) obj ).getStorageBytes() / KILO_SCALAR;
 
+		Optional<ItemStack> is;
+
 		switch( storage )
 		{
 			case 4:
-				for( final ItemStack stack : blocks.craftingStorage4k().maybeStack( 1 ).asSet() )
+				is = blocks.craftingStorage4k().maybeStack( 1 );
+				if( is.isPresent() )
 				{
-					return stack;
+					return is.get();
```

Comment:
Ah that was another `switch`. 

But here something like `is = blocks.craftingStorage16k().maybeStack( 1 ); break;` and then finally `return is.getOrElse(super);` would work and remove some LoC